tests/datasource/data_connector/test_configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,1576 @@
+        class_name: ConfiguredAssetAzureDataConnector
+        datasource_name: FAKE_DATASOURCE
+        name: TEST_DATA_CONNECTOR
+        default_regex:
+            pattern: alpha-(.*)\\.csv
+            group_names:
+                - index
+        container: my_container
+        name_starts_with: ""
+        assets:
+            alpha:
+        azure_options:
+            account_url: my_account_url.blob.core.windows.net
+            credential: my_credential
+    """,
+        return_mode="report_object",
+    )
+
+    assert report_object == expected_config_dict
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.list_azure_keys",
+    return_value=["alpha-1.csv", "alpha-2.csv", "alpha-3.csv"],
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+def test_instantiation_with_test_yaml_config_emits_proper_payload(
+    mock_azure_conn, mock_list_keys, mock_emit, empty_data_context_stats_enabled
+):
+    context: DataContext = empty_data_context_stats_enabled
+
+    context.test_yaml_config(
+        f"""
+        module_name: great_expectations.datasource.data_connector
+        class_name: ConfiguredAssetAzureDataConnector
+        datasource_name: FAKE_DATASOURCE
+        name: TEST_DATA_CONNECTOR
+        default_regex:
+            pattern: alpha-(.*)\\.csv
+            group_names:
+                - index
+        container: my_container
+        name_starts_with: ""
+        assets:
+            alpha:
+        azure_options:
+            account_url: my_account_url.blob.core.windows.net
+            credential: my_credential
+    """,
+        return_mode="report_object",
+    )
+    assert mock_emit.call_count == 1
+
+    anonymized_name = mock_emit.call_args_list[0][0][0]["event_payload"][
+        "anonymized_name"
+    ]
+    expected_call_args_list = [
+        mock.call(
+            {
+                "event": "data_context.test_yaml_config",
+                "event_payload": {
+                    "anonymized_name": anonymized_name,
+                    "parent_class": "ConfiguredAssetAzureDataConnector",
+                },
+                "success": True,
+            }
+        ),
+    ]
+    assert mock_emit.call_args_list == expected_call_args_list
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.list_azure_keys",
+    return_value=["alpha-1.csv", "alpha-2.csv", "alpha-3.csv"],
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+def test_instantiation_from_a_config_regex_does_not_match_paths(
```

Comment:
How come we have "does_not" in the method name?

Reply:
Modeled after a similar test in S3: https://github.com/great-expectations/great_expectations/blob/7036bb5c9476d3e7109c75b5571f4684ef0dc165/tests/datasource/data_connector/test_configured_asset_s3_data_connector.py#L183

The idea is that our `default_regex` does not capture the blobs in our container (hence `example_unmatched_data_references` containing a number of files. 

I do agree that this might be confusing. Any thoughts on a better name? Perhaps `test_instantiation_from_a_config_with_nonmatching_regex`?